### PR TITLE
use nanobind builtin stubgen

### LINKED ISSUE #855: Auto generated python stubs inconsistent with cpp API
This is a stellar project, I love it! I originally came here looking for a replacement for OpenSCAD's proprietary language. And I found a little gem :)

Now, here's a technical aspect I would like to hear the authors's / maintainer's thoughts on:

Following the stub generation from [bindings/python/README.md](https://github.com/elalish/manifold/blob/master/bindings/python/README.md) via `nanobind-stubgen manifold3d` I observe that the stubs don't properly reflect the API.

I do believe it is mostly a limitation of `nanobind`, however `manifold3d` can make an informed decision on whether to improve the situation.

My main issue with the stubs is that any IDE / linter / etc. will be confused, impairing the `manifold3d` python library user experience. For example, the `Manifold` cpp class has a few `static` constructors. Those are not marked as `@staticmethod` in `manifold.pyi`. Also, the operators for boolean combinations are not included in the the stub file (i.e. `__add__` etc. are missing).

I know that there are bindings for a couple of languages, and that this is not a first class python project. Still, solid stubs could greatly improve the developer experience when using the python package.

### PR DESCRIPTION
**Pull request by pca006132**
Closes #855.

The stub is not perfect because it is python, but I think we are pretty good now.

### CODE REVIEW

**Comment by pca006132** [comment]
@t-fi @wrongbad cannot request your review, so I will ping you instead.

File: bindings/python/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,8 @@ project(python)
 execute_process(
   COMMAND "${Python_EXECUTABLE}" -m nanobind --version
   OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE NB_VERSION)
-if(NB_VERSION VERSION_GREATER_EQUAL 1.8.0)
+# we are fine with 2.0.0
+if(NB_VERSION VERSION_GREATER_EQUAL 2.0.0)
```
Comment (elalish) [inline bindings/python/CMakeLists.txt:21]: Should we just go with 2.1.0 across the board?

File: bindings/python/examples/bricks.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def floor(length, width):
             )
         )
     if length == 1 and width > 1:
-        results.append(row(width - 1).rotate(0, 0, 90))
+        results.append(row(width - 1).rotate((0, 0, 90)))
```
Comment (elalish) [inline bindings/python/examples/bricks.py:87]: What are the extra parens for?

File: bindings/python/manifold3d.cpp [hunk 3]
```diff
@@ -327,7 +327,8 @@ NB_MODULE(manifold3d, m) {
       .def("refine", &Manifold::Refine, nb::arg("n"), manifold__refine__n)
       .def("refine_to_length", &Manifold::RefineToLength, nb::arg("length"),
            manifold__refine_to_length__length)
-      .def("to_mesh", &Manifold::GetMeshGL, nb::arg("normal_idx") = ivec3(0),
+      .def("to_mesh", &Manifold::GetMeshGL,
+           nb::arg("normal_idx") = std::make_tuple(0, 0, 0),
```
Comment (elalish) [inline bindings/python/manifold3d.cpp:331]: 👍

**Review by elalish — COMMENTED** [review]
Does this mean we can remove the python auto-gen docs now?

File: bindings/python/CMakeLists.txt (hunk 1 above)
Comment (pca006132) [reply bindings/python/CMakeLists.txt:21]: Some distros may have 2.1.0? Not sure about if we should support that.

**Comment by pca006132** [comment]
and no, we still need to autogen docs for getting the doc strings.

**Comment by pca006132** [comment]
will figure out the wasm issue tmr, probably some library version issue.

File: bindings/python/CMakeLists.txt (hunk 1 above)
Comment (elalish) [reply bindings/python/CMakeLists.txt:21]: Okay, so 2.0 is significantly more widely available than 2.1? This is fine if so - I was just thinking we could reduce CMake complexity a bit.

**Comment by pca006132** [comment]
Maybe we can just disable `build_nix` with js for now until I figure out what is going on, seems to be related to emscripten version.

**Comment by pca006132** [comment]
@elalish should we merge this?

**Review by elalish — APPROVED** [review]
Thanks!

**Pull request merged**